### PR TITLE
Logging improvements; several minor bptxm bugfixes

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably #6061) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably #6087) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/chains/evm/client/client.go [hunk 1]
```diff
@@ -41,10 +41,15 @@ type Client interface {
 	Call(result interface{}, method string, args ...interface{}) error
 	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
 	BatchCallContext(ctx context.Context, b []rpc.BatchElem) error
+	// BatchCallContextAll calls BatchCallContext for every single node including
+	// sendonlys.
+	// CAUTION: This should only be used for mass re-transmitting transactions, it
+	// might have unexpected effects to use it for anything else.
+	BatchCallContextAll(ctx context.Context, b []rpc.BatchElem) error
```
Comment (prashantkumar1982) [inline core/chains/evm/client/client.go:48]: Did you not intend to use this function from within common. batchSendTransactions()?

**Review by prashantkumar1982 — APPROVED** [review]

File: core/chains/evm/client/client.go (hunk 1 above)
Comment (samsondav) [reply core/chains/evm/client/client.go:48]: Good catch!

File: core/chains/evm/bulletprooftxmanager/common.go [hunk 2]
```diff
@@ -34,8 +34,10 @@ func batchSendTransactions(
 
 	reqs := make([]rpc.BatchElem, len(attempts))
```
Comment (prashantkumar1982) [inline core/chains/evm/bulletprooftxmanager/common.go:35]: Delete the comments in line 22-23. about the future improvement idea.
And thanks for implementing it  :)

**Review by prashantkumar1982 — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

File: core/chains/evm/client/pool_test.go [hunk 3]
```diff
@@ -275,3 +275,47 @@ func TestUnit_Pool_RunLoop(t *testing.T) {
...
+		s := new(evmmocks.SendOnlyNode)
+		s.Test(t)
+		s.On("BatchCallContext", ctx, b).Return(nil).Once()
+		sendonlys = append(sendonlys, s)
+		mockSendonlys = append(mockSendonlys, s)
+	}
+
+	p := evmclient.NewPool(logger.TestLogger(t), nodes, sendonlys, &cltest.FixtureChainID)
+
+	p.BatchCallContextAll(ctx, b)
```
Comment (jmank88) [inline core/chains/evm/client/pool_test.go:313]: Unchecked error?

File: core/chains/evm/client/pool_test.go [hunk 4]
```diff
@@ -275,3 +275,47 @@ func TestUnit_Pool_RunLoop(t *testing.T) {
...
+
+	nodeCount := 2
+	sendOnlyCount := 3
+
+	b := []rpc.BatchElem{
+		rpc.BatchElem{Method: "method", Args: []interface{}{1, false}},
+		rpc.BatchElem{Method: "method2"},
+	}
+
+	ctx := context.Background()
```
Comment (jmank88) [inline core/chains/evm/client/pool_test.go:293]: testutils.Context()?

**Pull request merged**